dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java

```diff
@@ -683,6 +674,46 @@ public Label getPaneItemsLabel() {
     public Label getCreateNewSessionLabel() {
         return createNewSessionLabel;
     }
+
+    private void startGetJob(final List<Ds3TreeTableValueCustom> listFiles,
+            final Path localPath) {
+        final ImmutableList<String> buckets = listFiles.stream()
```

Comment:
It looks like you don't need to create the bucket list.  If you removed the collect at the end of this stream and replaced it with the foreach, it will avoid the extra list creation.